### ISSUE TITLE
Cal indicator script- random events populating

### ISSUE THREAD

**Issue by himynamesned** [mvan231/Scriptable#4]
I’m getting some random calendar events populate in my widget. The colours for these events also don’t match any that I have (the events are all grey). 

I feel it has connected to some random calendar somehow. Any ideas?

Events such as “fair” and “New Year” on 11th Jan. doesn’t make sense. It’s also not pulling all events for calendars. As you can see, it has skipped some blue calendar events, but listed the next. 

not sure how I can debug. But could prove a privacy nightmare

**Comment by himynamesned** [comment]
![2604B42B-5BA6-47A3-95CB-7E83D82F0A29](https://user-images.githubusercontent.com/25020297/104172781-b21c5f80-53fc-11eb-918e-a8d7e5eea0ed.jpeg)
![81462941-967F-4CB0-A17E-B8BAE39F1EF8](https://user-images.githubusercontent.com/25020297/104172784-b3e62300-53fc-11eb-8f36-35abf7c8d5d0.jpeg)

**Comment by mvan231** [comment]
These seem like they are probably from the Siri suggestions calendar. I suggest ignoring that calendar during the setup or you'll have these issues. 

For some reason, events you've had suggested by Siri show up in the calendar event API even though they have not been accepted. I ran into this same problem before

**Comment by himynamesned** [comment]
Thanks but I’ve never been invited to the “fair”. Also looking directly at the Siri calendar, there are no events for the entire month. I also have “show declined events” turned on. 

So doesn’t really add up. 

I’ll run the set up again and ensure Siri is deselected and keep an eye out, but it doesn’t really add up.

**Comment by mvan231** [comment]
I can assure you there is nowhere else that it is coming from. I also had a random event show up on my calendar because Siri saw an event time for a "Dinner" at a certain time on a certain date and it became a Siri suggested calendar event which shows in the API in scriptable but not in the calendar app. Hopefully it will resolve for you by deselecting the Siri suggestions calendar

**Comment by mvan231** [comment]
Did that solve your issue?

**Comment by himynamesned** [comment]
Not sure. I’ve left it for now and no more events have populated. I’m waiting to see another random event to then toggle off the Siri calendar to confirm if that’s the source. I’ll give it a few more days and let you know. 

Thanks for follow up.

**Comment by mvan231** [comment]
Sounds good! One way you could try to force it is by having someone send you a message like "want to get dinner tonight at 7?" And it should see the date and item and add a Siri suggested event. But I tried doing this from myself and it won't work for some reason that way. But that's how I saw this happen before. 

My wife had sent me a message about meeting friends for dinner the next day and somehow it showed up in my Scriptable calendar events API response.

**Comment by mvan231** [comment]
Were you able to resolve it? If so, I'll go ahead and close this issue. Please let me know

**Comment by himynamesned** [comment]
To be honest it hasn’t reappeared despite tying to replicate with your Siri suggestion (but I also haven’t swapped my calendar sources). I’ll close for now. Thanks again.

**Comment by mvan231** [comment]
@himynamesned I just wanted to let you know that I figured it out where the events come from. It's a calendar called "Found in Natural Language". I'm going to reopen this issue until I can implement a good fix for it. 

I'm thinking it would be best if I reverse the logic and have the calendar picker be used for choosing the calendars you want displayed. This way that won't happen anymore. What do you think?

**Comment by mvan231** [comment]
This is fixed in v1.7 releasing soon

**Comment by himynamesned** [comment]
> @himynamesned I just wanted to let you know that I figured it out where the events come from. It's a calendar called "Found in Natural Language". I'm going to reopen this issue until I can implement a good fix for it.
> 
> I'm thinking it would be best if I reverse the logic and have the calendar picker be used for choosing the calendars you want displayed. This way that won't happen anymore. What do you think?

Sounds good. How did you find that out? Tbh, I’ve never understood why you’d want the logic to work the way it was set, so swapping is a food move. 

Thanks for following this through!

**Comment by mvan231** [comment]
It was a matter of how most people want to see their calendar app. Normally people selectively hide calendars rather than selectively show them. 

In any case, the logic in 1.7 will correct this behavior and the way I found this out was by seeing an event that was populated in the widget for me and then I checked the calendar name of that event in Scriptable. I always wondered why those events came into the API like that honestly
